src/parsing.jl

```diff
@@ -364,6 +360,39 @@ function Base.parse(::Type{Dates.CompoundPeriod}, pqv::PQValue{PQ_SYSTEM_TYPES[:
     return Dates.CompoundPeriod(periods)
 end
 
+## ranges
+_DEFAULT_TYPE_MAP[:int4range] = Interval{Int32}
+_DEFAULT_TYPE_MAP[:int8range] = Interval{Int64}
+_DEFAULT_TYPE_MAP[:numrange] = Interval{Decimal}
+_DEFAULT_TYPE_MAP[:tsrange] = Interval{DateTime}
+_DEFAULT_TYPE_MAP[:tstzrange] = Interval{ZonedDateTime}
+_DEFAULT_TYPE_MAP[:daterange] = Interval{Date}
+
+const RANGE_REGEX = r"^([\[\(])([^\[\(\]\),]+),([^\[\(\]\),]+)([\]\)])$"
```

Comment:
Not sure how others feel about this, but personally I think leaving comments around regex statements explaining what you're doing make it a lot easier if you ever have to come back to it.

Reply:
Another strategy is named capture groups

Reply:
Couldn't think of clear concise ways of naming the capture groups so went with comments.